base/docs/Docs.jl

```diff
@@ -320,7 +374,7 @@ function docm(ex)
     isexpr(ex, :->) && return docm(ex.args...)
     isexpr(ex, :call) && return findmethod(ex)
     isexpr(ex, :macrocall) && (ex = namify(ex))
-    :(doc($(esc(ex))))
+    :(doc(@var($(esc(ex)))))
```

Comment:
I think it makes sense to explicitly check for valid `@var` syntax here and fall back as before, otherwise you essentially disallow retreiving documentation for values.